### PR TITLE
Add :matches() BCD data

### PR DESCRIPTION
**Pull request by chrisdavidmills**
As per https://bugzilla.mozilla.org/show_bug.cgi?id=1440868

### CODE REVIEW

File: css/selectors/matchesfunction.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
+  "css": {
+    "selectors": {
+      "matchesfunction": {
```
Comment (teoli2003) [inline css/selectors/matchesfunction.json]: I would call it "matches" as there is no ambiguity.

File: css/selectors/matchesfunction.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        "__compat": {
+          "description": "<code>:matches()</code>",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/CSS/:matches()",
+          "support": {
+            "webview_android": {
+              "version_added": "66",
+              "notes": "Equivalent functionality also available via the <code>:-webkit-any()</code> pseudo-class."
```
Comment (teoli2003) [inline css/selectors/matchesfunction.json]: I would merge it with :any():
[{
 "version_added": "66"
},
{
  "version_added": "12",  (value to be double-checked on :any() page)
  "alternate_name": "-webkit-any()"
}]

File: css/selectors/matchesfunction.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            },
+            "chrome": {
+              "version_added": "66",
+              "notes": "Equivalent functionality also available via the <code>:-webkit-any()</code> pseudo-class."
```
Comment (teoli2003) [inline css/selectors/matchesfunction.json]: Same here

File: css/selectors/matchesfunction.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            },
+            "chrome_android": {
+              "version_added": "66",
+              "notes": "Equivalent functionality also available via the <code>:-webkit-any()</code> pseudo-class."
```
Comment (teoli2003) [inline css/selectors/matchesfunction.json]: Same here

File: css/selectors/matchesfunction.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            },
+            "edge": {
+              "version_added": false
+            },
+            "edge_mobile": {
+              "version_added": false
+            },
+            "firefox": {
+              "version_added": false,
+              "notes": "Equivalent functionality available via the <code>:-moz-any()</code> pseudo-class."
```
Comment (teoli2003) [inline css/selectors/matchesfunction.json]: Same here

File: css/selectors/matchesfunction.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            },
+            "firefox_android": {
+              "version_added": false,
+              "notes": "Equivalent functionality available via the <code>:-moz-any()</code> pseudo-class."
```
Comment (teoli2003) [inline css/selectors/matchesfunction.json]: Same here

File: css/selectors/matchesfunction.json [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": false,
+              "notes": "Equivalent functionality available via the <code>:-webkit-any()</code> pseudo-class."
```
Comment (teoli2003) [inline css/selectors/matchesfunction.json]: Same here

File: css/selectors/matchesfunction.json [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+            },
+            "opera_android": {
+              "version_added": false,
+              "notes": "Equivalent functionality available via the <code>:-webkit-any()</code> pseudo-class."
```
Comment (teoli2003) [inline css/selectors/matchesfunction.json]: Same here

**Comment by chrisdavidmills** [comment]
I believe this is all done now. I've also updated the matches page, and redirected the any page to it:

https://developer.mozilla.org/en-US/docs/Web/CSS/:matches()

**Comment by ExE-Boss** [comment]
@chrisdavidmills I have fixed the broken JSON errors at chrisdavidmills/browser-compat-data#1.

**Comment by chrisdavidmills** [comment]
Ooops, I forgot to check for errors ;-)

I will change the brackets on the matches page right away.

The reason for this confusion is that I've come across a collision. In the shadow DOM spec, you've got these two pseudo-classes:

* https://developer.mozilla.org/en-US/docs/Web/CSS/:host
* https://developer.mozilla.org/en-US/docs/Web/CSS/:host()

I am wondering what we should do in such a case. Any thoughts @Elchi3 or @teoli2003 ?

**Comment by ExE-Boss** [comment]
One of the issues with the brackets is that linking a page with brackets in them on BMO results in the brackets not being included in the `<a>` element.
bmo-broken-url

File: css/selectors/matches.json [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+          "description": "<code>:matches()</code>",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/CSS/:matches()",
+          "support": {
+            "webview_android": [
+              {
+                "version_added": "66",
+              },
+              {
+                "version_added": "12",
+                "alternate_name": ":-webkit-any()"
```
Comment (teoli2003) [inline css/selectors/matches.json]: No ()

File: css/selectors/matches.json [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+              }
+            ],
+            "chrome": [
+              {
+                "version_added": "66",
+              },
+              {
+                "version_added": "12",
+                "alternate_name": ":-webkit-any()"
```
Comment (teoli2003) [inline css/selectors/matches.json]: No ()

File: css/selectors/matches.json [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+              }
+            ],
+            "chrome_android": [
+              {
+                "version_added": "66",
+              },
+              {
+                "version_added": "12",
+                "alternate_name": ":-webkit-any()"
```
Comment (teoli2003) [inline css/selectors/matches.json]: No ()

File: css/selectors/matches.json [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+            "edge_mobile": {
+              "version_added": false
+            },
+            "firefox": [
+              {
+                "version_added": false,
+              },
+              {
+                "version_added": "4",
+                "alternate_name": ":-moz-any()"
```
Comment (teoli2003) [inline css/selectors/matches.json]: No ()

File: css/selectors/matches.json [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+              }
+            ],
+            "firefox_android": [
+              {
+                "version_added": false,
+              },
+              {
+                "version_added": "4",
+                "alternate_name": ":-moz-any()"
```
Comment (teoli2003) [inline css/selectors/matches.json]: No ()

File: css/selectors/matches.json [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+            "ie": {
+              "version_added": false
+            },
+            "opera": [
+              {
+                "version_added": false,
+              },
+              {
+                "version_added": true,
+                "alternate_name": ":-webkit-any()"
```
Comment (teoli2003) [inline css/selectors/matches.json]: No ()

File: css/selectors/matches.json [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+              }
+            ],
+            "opera_android": [
+              {
+                "version_added": false,
+              },
+              {
+                "version_added": true,
+                "alternate_name": ":-webkit-any()"
```
Comment (teoli2003) [inline css/selectors/matches.json]: No ()

File: css/selectors/matches.json [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+              }
+            ],
+            "safari": [
+              {
+                "version_added": "9",
+              },
+              {
+                "version_added": "5",
+                "alternate_name": ":-webkit-any()"
```
Comment (teoli2003) [inline css/selectors/matches.json]: No ()

File: css/selectors/matches.json [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+              }
+            ],
+            "safari_ios": [
+              {
+                "version_added": "9",
+              },
+              {
+                "version_added": "5",
+                "alternate_name": ":-webkit-any()"
```
Comment (teoli2003) [inline css/selectors/matches.json]: No ()

File: css/selectors/matches.json [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+            ],
+            "edge": {
+              "version_added": false
+            },
+            "edge_mobile": {
+              "version_added": false
+            },
+            "firefox": [
+              {
+                "version_added": false
```
Comment (ExE-Boss) [inline css/selectors/matches.json]: This shouldn’t be here because no other file has this.

File: css/selectors/matches.json [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+              },
+              {
+                "version_added": "4",
+                "alternative_name": ":-moz-any"
+              }
+            ],
+            "firefox_android": [
+              {
+                "version_added": false
```
Comment (ExE-Boss) [inline css/selectors/matches.json]: ditto

File: css/selectors/matches.json [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+                "version_added": "4",
+                "alternative_name": ":-moz-any"
+              }
+            ],
+            "ie": {
+              "version_added": false
+            },
+            "opera": [
+              {
+                "version_added": false
```
Comment (ExE-Boss) [inline css/selectors/matches.json]: ditto

File: css/selectors/matches.json [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+              },
+              {
+                "version_added": true,
+                "alternative_name": ":-webkit-any"
+              }
+            ],
+            "opera_android": [
+              {
+                "version_added": false
```
Comment (ExE-Boss) [inline css/selectors/matches.json]: ditto

**Comment by chrisdavidmills** [comment]
Updated - you mean like this?

**Pull request merged**